src/state-manager/TransactionQueue.ts

```diff
@@ -5536,35 +6112,61 @@ class TransactionQueue {
                 continue
               }
 
-              if (
-                this.queueTimingFixes === true &&
-                this.processQueue_accountSeen(seenAccounts, queueEntry) === true
-              ) {
-                //we are stuck in line so no cause to ask for data yet.
+              // This code is wrong, so disabling it for now
+              // if (
+              //   this.queueTimingFixes === true &&
+              //   this.processQueue_accountSeen(seenAccounts, queueEntry) === true
+              // ) {
+              //   //we are stuck in line so no cause to ask for data yet.
 
-                //TODO may need a flag that know if a TX was stuck until time m.. then let it not
-                //ask for other accoun data right away...
-                nestedCountersInstance.countEvent(`processing`, `awaiting data. stuck in line`)
-                continue
+              //   //TODO may need a flag that know if a TX was stuck until time m.. then let it not
+              //   //ask for other accoun data right away...
+
+              //   //This counter seems wrong.  the processQueue_accountSeen is just detecting where we 
+              //   // called processQueue_markAccountsSeen before 
+              //   nestedCountersInstance.countEvent(`processing`, `awaiting data. stuck in line - but not really`)
+              //   continue
+              // }
+
+              //TODO check for receipt and move to repair state / await final data
+
+              if(this.config.stateManager.awaitingDataCanBailOnReceipt){
+                const receipt = this.stateManager.getReceipt2(queueEntry)
+                if(receipt != null){
+                  //we saw a receipt so we can move to await final data
+
+                  this.updateTxState(queueEntry, 'await final data', 'receipt while waiting for initial data')
+                  continue
```

Comment:
so this can cause majority of nodes to go to 'await final data' state

Reply:
I think this cant impact a majority because, a receipt now requires a supermajority of nodes  (with POQo)
this helps a node that was stuck waiting for data while 2/3rds+ were able to get the data and form a receipt.

Reply:
I added a counter so we can track how often this is activated.  going to start tracking a lot of things like this in automation